src/ophyd_async/core/device.py

```diff
@@ -83,12 +91,21 @@ async def connect(self, sim: bool = False, timeout: float = DEFAULT_TIMEOUT):
         timeout:
             Time to wait before failing with a TimeoutError.
         """
-        coros = {
-            name: child_device.connect(sim, timeout=timeout)
-            for name, child_device in self.children()
-        }
-        if coros:
-            await wait_for_connection(**coros)
+        previous_connect_ok = (
+            self._connect_task
+            and self._connect_task.done()
+            and not self._connect_task.exception()
+        )
+        if force_reconnect or not previous_connect_ok:
+            # Kick off a connection
+            coros = {
+                name: child_device.connect(sim, timeout=timeout)
+                for name, child_device in self.children()
+            }
+            self._connect_task = asyncio.create_task(wait_for_connection(**coros))
```

Comment:
Could we please have a test running this without the plan stub:

1. We try to connect a `Device` which doesn't work (giving one of its signals a hard backend should work)
2. It doesn't work so then we allow the signal to have a sim backend
3. We try to connect again now that the sim backend is there, so it works